docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -0,0 +1,129 @@
+            this.last = last;
+        }
+    }
+
+    @Path("/person")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public class PersonResource {
+
+        private final Jsonb jsonb;
+
+        public PersonResource() {
+            jsonb = JsonbBuilder.create(new JsonbConfig());
+        }
+
+        @GET
+        public Response list() {
+            return Response.ok(jsonb.fromJson("{\"first\":  \"foo\", \"last\":  \"bar\"}", Person.class)).build();
+        }
+    }
+----
+
+If we were to use the code above, we would get an exception like the following when using the native image:
+
+[source]
+----
+ Exception handling request to /person: org.jboss.resteasy.spi.UnhandledException: javax.json.bind.JsonbException: Can't create instance of a class: class org.acme.jsonb.Person, No default constructor found
+----
+
+To get around the problem, simply annotate `Person` with `@RegisterForReflection` so Quarkus will register the class with GraalVM as a class using reflection.
+
+== Including resources
+
+By default when building a native image, GraalVM will not include any of the resources which are on the classpath into the native image it creates. Such resources that are meant to be part of the native image need to be specified explicitly.
+
+If one were to execute the native image generation process manually, the `-H:IncludeResources=` flag would be used to point to a json configuration file that specifies which resources are to be included.
+
+Quarkus eliminates the need for such a json configuration file by allowing extension authors to specify a `SubstrateResourceBuildItem`.
+
+For more information about GraalVM's resource handling in native images please see https://github.com/oracle/graal/blob/master/substratevm/RESOURCES.md[this].
+
+
+== Delay class initialization
+
+There are cases were the initialization of certain classes which is done in a static block (which means by default that GraalVM performs the initialization at image build time) needs to be postponed to runtime.
+Typically omitting such configuration would result in a runtime exception like the following:
+
+[source]
+----
+Error: No instances are allowed in the image heap for a class that is initialized or reinitialized at image runtime: sun.security.provider.NativePRNG
+Trace: object java.security.SecureRandom
+method com.amazonaws.services.s3.model.CryptoConfiguration.<init>(CryptoMode)
+Call path from entry point to com.amazonaws.services.s3.model.CryptoConfiguration.<init>(CryptoMode):
+----
+
+If one were to execute the native image generation process manually, the `--delay-class-initialization-to-runtime=com.amazonaws.services.s3.model.CryptoConfiguration` flag would need to be passed to `native-image` to configure runtime initialization for `CryptoConfiguration`.
+
+Quarkus simplifies things by allowing extensions authors to simply register a `RuntimeInitializedClassBuildItem` like so:
```

Comment:
Same remark about the extension authors part and where I would have to put this `@BuildStep` thing.